cmds/dhclient/dhcp6client/packetSock.go

```diff
@@ -0,0 +1,159 @@
+package dhcp6client
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/mdlayher/dhcp6"
+	"golang.org/x/sys/unix"
+)
+
+type packetSock struct {
+	fd      int
+	ifindex int
+}
+
+var bcastMAC = []byte{255, 255, 255, 255, 255, 255}
+
+func NewPacketSock(ifindex int) (*packetSock, error) {
```

Comment:
I think you have to decide what abstraction you want this to be.

It's either:
```go
type dhcp6packetsocket interface {
  WritePacket(dhcp6.Packet) error
  ReadPacket() (dhcp6.Packet, error)
}
```

/or/

```go
type packetSocket interface {
  Write(p []byte) (int, error)
  Read(p []byte) (int, error)
}
```

Either this consumes and produces packets, or it produces and consumes bytes and the layer above does the marshaling.